cgsmiles/pysmiles_utils.py

```diff
@@ -3,6 +3,49 @@
 VALENCES = pysmiles.smiles_helper.VALENCES
 VALENCES.update({"H": (1,)})
 
+def _smiles_node_iter(smiles_str):
+    """
+    Iterate over all nodes in SMILES string and return
+    the index of the node.
+    """
+    organic_subset = 'B C N O P S F Cl Br I * b c n o s p'.split()
+    batom = False
+    for idx, node in enumerate(smiles_str):
+        if node == '[':
+            batom = True
+            start = idx
+
+        if node == ']' and batom:
+            stop = idx+1
+            batom = False
+            yield start, stop
+
+        if node in organic_subset and not batom:
+            yield idx, idx + 1
+
+def strip_aromatic_nodes(smiles_str):
```

Comment:
I don't like the name of the function very much, since it deals with the smiles string, and not with graph nodes